debugutils/aggregator.go

```diff
@@ -0,0 +1,109 @@
+package debugutils
+
+import (
+	"io"
+	"path/filepath"
+
+	"github.com/solo-io/go-utils/errors"
+	"github.com/solo-io/go-utils/installutils/helmchart"
+	"github.com/solo-io/go-utils/tarutils"
+	"github.com/spf13/afero"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	aggregatorName = "aggregator"
+)
+
+type Aggregator struct {
+	resourceCollector ResourceCollector
+	logCollector      LogCollector
+	storageClient     StorageClient
+	fs                afero.Fs
+
+	dir string
+}
+
+func NewAggregator(resourceCollector ResourceCollector, logCollector LogCollector, storageClient StorageClient, fs afero.Fs, dir string) *Aggregator {
+	return &Aggregator{resourceCollector: resourceCollector, logCollector: logCollector, storageClient: storageClient, fs: fs, dir: dir}
+}
+
+func DefaultAggregator() (*Aggregator, error) {
+	fs := afero.NewOsFs()
+	storageClient := NewFileStorageClient(fs)
```

Comment:
Seems like we should be using the same storage client in the resource and log collector